tests/core/pyspec/eth2spec/test/phase0/unittests/fork_choice/test_on_block.py

```diff
@@ -166,20 +61,28 @@ def test_on_block_finalized_skip_slots_not_in_skip_chain(spec, state):
     transition_unsigned_block(spec, state, block)
     block.state_root = state.hash_tree_root()
     store = spec.get_forkchoice_store(state, block)
-    store.finalized_checkpoint = spec.Checkpoint(
-        epoch=store.finalized_checkpoint.epoch + 2,
-        root=store.finalized_checkpoint.root
-    )
 
-    # First transition through the epoch to ensure no skipped slots
-    state, store, _ = apply_next_epoch_with_attestations(spec, state, store)
+    current_time = state.slot * spec.config.SECONDS_PER_SLOT + store.genesis_time
+    spec.on_tick(store, current_time)
+    assert store.time == current_time
+
+    pre_finalized_checkpoint_epoch = store.finalized_checkpoint.epoch
+
+    # Finalized
+    for _ in range(3):
+        state, store, _ = yield from apply_next_epoch_with_attestations(spec, state, store)
+    assert store.finalized_checkpoint.epoch == pre_finalized_checkpoint_epoch + 1
 
     # Now build a block at later slot than finalized epoch
     # Includes finalized block in chain, but not at appropriate skip slot
-    block = build_empty_block(spec, state, spec.compute_start_slot_at_epoch(store.finalized_checkpoint.epoch) + 2)
-    signed_block = state_transition_and_sign_block(spec, state, block)
+    pre_state = store.block_states[block.hash_tree_root()]
+    block = build_empty_block(spec,
+                              state=pre_state,
+                              slot=spec.compute_start_slot_at_epoch(store.finalized_checkpoint.epoch) + 2)
+    signed_block = sign_block(spec, pre_state, block)
+
     spec.on_tick(store, store.time + state.slot * spec.config.SECONDS_PER_SLOT)
```

Comment:
*(not relevant to your work)*
This should be `store.genesis_time + state.slot * spec.config.SECONDS_PER_SLOT`

but beacuse we don't want time to matter in this, overshooting is actually totally fine... so maybe no change